### DISCUSSION TITLE
[Select] setting required attribute

### DISCUSSION THREAD

**Discussion by MadadiSaeed** [chakra-ui/zag#976] [General]
Hi. There is one improvement related to Select component:

Adding `required` attribute to the hidden select element when we add `required` property to Select component.

**Question**: In regard to Select component when I put Select component inside a form and submit the form to the server without any first interaction with Select component or without any option selection or without setting any `value` or `defaultValue` why is that I receive the first option on the server?

**Comment by segunadebayo** [comment]
Hi @samadadi,

- required attr: Feel free to add this to the hidden select. Adding this will require a change in the other machines for consistency. Not willing to do that right now.
- value selection: This might be a side-effect of using the hidden select. I recommend you render an option with an empty value first before rendering the items.

```jsx
<select {...api.hiddenSelectProps}>
   <option value="" />
   {...}
</select>
```

**Comment by segunadebayo** [comment]
Closing as outdated — the select API has been reworked in v1.0.
